File: C++/Primes/primeList.cpp
#include "pch.h"

#include <vector>

vector<int> primeList(int maxN)
{
    vector<bool> boolL(maxN + 1, true);
    boolL[0] = boolL[1] = false;
    for (int i = 2; i < maxN; ++i) {
        if (boolL[i]) {
            for (int j = i + i; j < maxN; j += i) {
                boolL[j] = false;
            }
        }
    }
    vector<int> r;
    for (int i = 2; i < maxN; ++i) if (boolL[i]) r.push_back(i);
    return r;
}

int main()
{
    cout << primeList(2000000).size() << endl;
}

File: Python/Primes/primeList.py

def primeList(maxN):
    l = [True for i in range(maxN + 1)]
    l[0] = l[1] = False
    for i in range(2, maxN):
        if not l[i]: pass
        else:
            for j in range(2*i,maxN,i):
                l[j] = False
    return [i for (i,b) in zip(range(2,maxN), l[2:]) if b]

print primeList(100)
